pkg/common/cliprinter/internal/protojson/protojson.go

```diff
@@ -34,13 +34,62 @@ func Print(msgs []proto.Message, stdout, stderr io.Writer) error {
 
 		jms = append(jms, jb)
 	}
-
 	var err error
-	if len(jms) == 1 {
-		err = json.NewEncoder(stdout).Encode(jms[0])
+
+	parsedJms, err := parseJSONMessages(jms)
+	if err != nil {
+		_ = errorjson.Print(err, stdout, stderr)
+		return err
+	}
+
+	if len(parsedJms) == 1 {
+		err = json.NewEncoder(stdout).Encode(parsedJms[0])
 	} else {
-		err = json.NewEncoder(stdout).Encode(jms)
+		err = json.NewEncoder(stdout).Encode(parsedJms)
 	}
 
 	return err
 }
+
+func parseJSONMessages(jms []json.RawMessage) ([]json.RawMessage, error) {
+	var parsedJms []json.RawMessage
+	for _, jm := range jms {
+		parsedJm, err := parseJSONMessage(jm)
+		if err != nil {
+			return nil, err
+		}
+		parsedJms = append(parsedJms, parsedJm)
+	}
+
+	return parsedJms, nil
+}
+
+func parseJSONMessage(jm json.RawMessage) (json.RawMessage, error) {
+	var jmMap map[string]interface{}
+	if err := json.Unmarshal(jm, &jmMap); err != nil {
+		return nil, err
+	}
+
+	removeNulls(jmMap)
+
+	return json.Marshal(jmMap)
+}
+
+func removeNulls(jsonMap map[string]interface{}) {
+	for key, val := range jsonMap {
+		if val == nil {
+			delete(jsonMap, key)
+			continue
+		}
+		switch v := val.(type) {
+		case map[string]interface{}:
+			removeNulls(v)
+		case []interface{}:
+			for _, elem := range v {
+				if e, ok := elem.(map[string]interface{}); ok {
```

Comment:
Is it possble that the element in the interface array is a type other than the map? ie can it be a bool?

Reply:
Yes, but the most important here is that it can have nil values or []interface{} values. I'm gonna update it to include the logic for this.

Reply:
got it, the other types are ok to leave as is.

Reply:
[f05c578](https://github.com/spiffe/spire/pull/3715/commits/f05c578175592723d32170714bfc86822ad5b113)